subsys/ipc/ipc_service/lib/icmsg.c

```diff
@@ -34,10 +35,30 @@ static int mbox_deinit(const struct icmsg_config_t *conf,
 	}
 
 	(void)k_work_cancel(&dev_data->mbox_work);
+	(void)k_work_cancel_delayable(&dev_data->bond_notify_work);
 
 	return 0;
 }
 
+static void bond_notify_process(struct k_work *item)
+{
+	int ret;
+	struct k_work_delayable *dwork = k_work_delayable_from_work(item);
+	struct icmsg_data_t *dev_data =
+		CONTAINER_OF(dwork, struct icmsg_data_t, bond_notify_work);
+
+	ret = mbox_send(&dev_data->cfg->mbox_tx, NULL);
+	__ASSERT_NO_MSG(ret >= 0);
+	(void)ret;
```

Comment:
I think we should retry with same timeout if this `mbox_send` fails. What do you think?

Reply:
Lets retry. I think we would get an information that something is wrong by bonding timeout anyway.

Reply:
Well, yeah. Please don't add this retry i was wondering if that's necessary but i think its now.